File: src/main/java/com/bitwise/pizzaShop/database/CrustDatabase.java
package com.bitwise.pizzaShop.database;

import java.util.HashMap;

import com.bitwise.pizzaShop.pizza.Crust;

public class CrustDatabase {
	private static HashMap<String, Crust> pizzaCrusts;
	
	static{
		pizzaCrusts = new HashMap<String, Crust>();
		pizzaCrusts.put("Pan Pizza", new Crust("Pan Pizza", 100));
		pizzaCrusts.put("Thin Crust", new Crust("Thin Crust", 150));
		pizzaCrusts.put("Regular", new Crust("Regular", 0));
		pizzaCrusts.put("Cheese Burst", new Crust("Cheese Burst", 200));
	}

	public static HashMap<String, Crust> getPizzaCrusts() {
		return pizzaCrusts;
	}

	public static void setPizzaCrusts(HashMap<String, Crust> pizzaCrusts) {
		CrustDatabase.pizzaCrusts = pizzaCrusts;
	}
}

File: src/main/scala/aniruddha/dominos/databases/CrustDatabase.scala
package aniruddha.dominos.databases

import aniruddha.dominos.basic_ingredients.Crust

import scala.collection.immutable.HashMap

/**
  * Created by AniruddhaS on 6/27/2016.
  */
class CrustDatabase {
  val crusts:Map[String,Crust]=
    HashMap("Pan Pizza"->Crust("Pan Pizza",100),
      ("Thin Crust"->Crust("Thin Crust",150)),
      ("Regular"->Crust("Regular",0)),
      ("Cheese Burst"->Crust("Cheese Burst",200))
  )
  def getPrice(s:String):Double= crusts(s).getPrice


  def canEqual(other: Any): Boolean = other.isInstanceOf[CrustDatabase]

  override def equals(other: Any): Boolean = other match {
    case that: CrustDatabase =>
      (that canEqual this) &&
        crusts == that.crusts
    case _ => false
  }

  override def hashCode(): Int = {
    val state = Seq(crusts)
    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
  }
}
